www/localPackage.ts

```diff
@@ -71,41 +71,85 @@ class LocalPackage extends Package implements ILocalPackage {
                 }
             };
 
             FileUtil.getDataDirectory(LocalPackage.DownloadUnzipDir, false, (error: Error, directoryEntry: DirectoryEntry) => {
-                var unzipPackage = () => {
+                var unzipAndVerifyPackage = () => {
                     FileUtil.getDataDirectory(LocalPackage.DownloadUnzipDir, true, (innerError: Error, unzipDir: DirectoryEntry) => {
                         if (innerError) {
                             installError && installError(innerError);
-                        } else {
-                            zip.unzip(this.localPath, unzipDir.toInternalURL(), newPackageUnzipped);
+                            return;
                         }
+
+                        zip.unzip(this.localPath, unzipDir.toInternalURL(), (unzipError: any) => {
+                            if (unzipError) {
+                                installError && installError(new Error("Could not unzip package. " + CodePushUtil.getErrorMessage(unzipError)));
+                            }
+                           this.verifyPackage(unzipDir, installError, newPackageUnzippedAndVerified);
+                        });
+
                     });
                 };
 
                 if (!error && !!directoryEntry) {
                     /* Unzip directory not clean */
                     directoryEntry.removeRecursively(() => {
-                        unzipPackage();
+                        unzipAndVerifyPackage();
                     }, (cleanupError: FileError) => {
                         installError && installError(FileUtil.fileErrorToError(cleanupError));
                     });
                 } else {
-                    unzipPackage();
+                    unzipAndVerifyPackage();
                 }
             });
         } catch (e) {
             installError && installError(new Error("An error occured while installing the package. " + CodePushUtil.getErrorMessage(e)));
         }
     }
 
+    private verifyPackage(unzipDir: DirectoryEntry, installError: ErrorCallback, callback: Callback<boolean>): void {
+        var packageHashSuccess = (localHash: string) => {
+            FileUtil.readFile(cordova.file.dataDirectory, unzipDir.fullPath + '/www', '.codepushrelease', (error, contents) => {
+                var verifySignatureSuccess = (expectedHash?: string) => {
+                    // first, we always compare the hash we just calculated to the packageHash reported from the server
+                    if (localHash !== this.packageHash) {
+                        installError(new Error("package hash verification failed"));
+                        return;
+                    }
+
+                    // this happens if (and only if) no public key is available in config.xml
+                    // -> no code signing
+                    if (!expectedHash) {
+                        callback(null, false);
```

Comment:
We should log  `The update contents succeeded the data integrity check.` for consistency with RN code signing implementation.